### PR TITLE
Migrate to using windows-2022 pool

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

File: eng/pipelines/eng-common-sync.yml [hunk 1]
```diff
@@ -92,6 +92,7 @@ stages:
           environment: githubmerges
 
           pool:
+            name: azsdk-pool-mms-win-2022-general
```
Comment (benbp) [inline eng/pipelines/eng-common-sync.yml:95]: Can we try switching to ubuntu here?

**Review by benbp — APPROVED** [review]

File: eng/pipelines/eng-common-sync.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/eng-common-sync.yml:95]: Yes it theory but we would need to test. We did move the first job to ubuntu.

**Comment by konrad-jamrozik** [comment]
This is related to:
- #3407

**Pull request merged**